packages/adapter-components/src/client/http_client.ts

```diff
@@ -231,15 +238,15 @@ export abstract class AdapterHTTPClient<
         }
         : undefined
 
-      const res = isMethodWithData(params)
+      const res = isMethodWithDataParam(method)
         ? await this.apiClient[method](
           url,
-          params.data,
+          isMethodWithData(params) ? params.data : undefined,
```

Comment:
nit: can't we just call 
```suggestion
          params.data,
```
? if it doesn't exist it will be undefined anyway, no?

Reply:
typescript doesn't allow it as params might be `ClientBaseParams` that doesn't contain `data`